vars/parseStageInfo.groovy

```diff
@@ -70,6 +70,21 @@ String get_commit_pragma_tags(String pragma_suffix) {
   return pragma_tag
 }
 
+String get_ftest_arg(String build_param, String commit_pragma, String pragma_suffix) {
+  // Get the ftest argument from either the build parameter or commit pragma variables
+  String ftest_arg
+
+  // Highest ftest argument priority is build parameter
+  if (build_param && build_param != '') {
+    ftest_arg = build_param
+  } else {
+    // Next highest priority is a stage specific pragma, followed by the common pragma
+    ftest_arg = cachedCommitPragma(
+      commit_pragma + pragma_suffix, cachedCommitPragma(commit_pragma, null))
+  }
+  return ftest_arg
```

Comment:
This could be more concisely written as:

```suggestion
  // Highest ftest argument priority is build parameter, followed by stage specific pragma, followed by the common pragma
  return build_param ?: cachedCommitPragma(commit_pragma + pragma_suffix,
                                           cachedCommitPragma(commit_pragma, ''))
```

But then becomes questionable whether it needs to be wrapped in a function.